src/qibolab/instruments/qblox.py

```diff
@@ -1101,12 +1101,15 @@ def acquire(self):
                     self.device.store_scope_acquisition(sequencer.number, "scope_acquisition")
                     scope_acquisition_raw_results = self.device.get_acquisitions(sequencer.number)["scope_acquisition"]
 
+        acquisition_results["demodulated_integrated_averaged"] = {}
+        acquisition_results["averaged_raw"] = {}
+        acquisition_results["averaged_demodulated_integrated"] = {}
+        acquisition_results["demodulated_integrated_binned"] = {}
+        acquisition_results["demodulated_integrated_classified_binned"] = {}
+        acquisition_results["probability"] = {}
```

Comment:
Hi @stavros11, the fact that the acquisition returns different dictionaries based on whether demodulation is done in software or hardware, was intentional (although it may have been a poor design decision).
This change was not really necessary to fix the bug, and I would have preferred to park it until we replaced results dictionaries with objects, but I guess it won't break anything.